### PR TITLE
Fix authority measure calculation for PCR[7]

### PR DESCRIPTION
**Pull request by sean-m-mcginn**
This change corrects the authority measure calculation in the common BootGuardTpmEventLogLib

If BtG KM is signed using RSA 3K key, a SHA384 hash should be added to the authority PCR structure

If BtG KM is signed using RSA 2K key, a SHA256 hash (padded to 384 bits) should be added to the authority PCR structure

The entire authority PCR structure should be initialized to zero before being populated

### CODE REVIEW

**Comment by cans-arista** [comment]
Thanks for the fix!
I tested this patch on an ICX-D platform with BtG profile 3 and Authority Data Measurement enabled. PCR 7 value can be reproduced using the log generated by Slimboot.
Tested the changed event type for Authority Data Measurement event with tpm2_eventlog (included in tpm2-tools (5.2-1build1)). The PCR 7 event TCG log can be decoded with the tool. (Previously it would fail with event data not correct)
(Fix #2013)

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c [hunk 1]
```diff
@@ -432,7 +431,7 @@ typedef struct {
 typedef struct {
   UINT64 AcmPolicySts;
   UINT16 AcmSvn;                               // ACM_SVN from ACM Header
-  UINT8  AcmKeyHash[SHA384_DIGEST_SIZE];       // The hash of the key used to verify the ACM (SHAxxx)
+  UINT8  AcmKeyHash[SHA256_DIGEST_SIZE];       // The hash of the key used to verify the ACM (SHAxxx)
```
Comment (lsubash) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c:434]: note, newer client platforms has ACH keys hash SHA384

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c [hunk 2]
```diff
@@ -1166,35 +1178,41 @@ CreateDetailPcrEvent (
...
-      }
+    NewEventHdr.PCRIndex  = 7;
+    NewEventHdr.EventType = EV_PLATFORM_CONFIG_FLAGS;
+    CaculateAuthorityPCRExtendValue (Sha256);
+
+    ZeroMem (Digests, sizeof(TPML_DIGEST_VALUES));
+    if ((ActivePcrBanks & HASH_ALG_SHA256) != 0) {
+      Digests->digests[Digests->count].hashAlg = TPM_ALG_SHA256;
+      CopyMem (Digests->digests[Digests->count].digest.sha256, Sha256, SHA256_DIGEST_SIZE);
+      Digests->count++;
```
Comment (lsubash) [inline Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c:1205]: We have to support for SHA384 and SM3 also.

File: Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c (hunk 1 above)
Comment (lsubash) [reply Silicon/CommonSocPkg/Library/BootGuardLibCBnT/BootGuardTpmEventLogLib.c:434]: Yes, this lib should not have issue.

**Review by gdong1 — APPROVED** [review]

**Pull request merged**